### PR TITLE
NetworkId parameter to start multiple Tapyrus-networks 

### PR DESCRIPTION
**Pull request by Naviabheeman**
Added "networkid" command line option to configure separate network id.
 - network magic bytes are calculated based on this network id
 - datadir name depends on this parameter: it is now `<chain>-<networkid> `and genesis file name is `genesis.<networkid>`

Refactoring:
CChainParams is all constants (different but fixed for main and regtest modes.)
CBaseChainParams contains all the network parameters - magic bytes, genesis blocks, seeds and datadir etc. This PR does not make seeds configurable. I'll add it in the next PR.

Removed -testnet and -paradium modes.
Moved tapyrus modes to new header and defined default network Ids for two modes - MAIN and REGTEST.

Updated makefile - changed target binaries and link binaries to resolve build errors 
Updated unit tests and functional tests to use networkid parameter.

### CODE REVIEW

File: src/test/util_tests.cpp [hunk 1]
```diff
@@ -542,80 +542,56 @@ BOOST_AUTO_TEST_CASE(util_GetArg)
...
-BOOST_AUTO_TEST_CASE(util_GetChainName)
+BOOST_AUTO_TEST_CASE(util_GetChainMode)
 {
     TestArgsManager test_args;
-    const char* avail_args[] = {"-testnet", "-regtest"};
-    test_args.SetupArgs(2, avail_args);
+    const char* avail_args[] = {"-regtest"};
+    test_args.SetupArgs(1, avail_args);
 
     const char* argv_testnet[] = {"cmd", "-testnet"};
```
Comment (azuchi) [inline src/test/util_tests.cpp]: Since `-testnet` parameter was removed, is it correct to not specify it?

File: doc/tapyrus/getting_started.md [hunk 2]
```diff
@@ -12,7 +12,6 @@ Tapyrus network has different kinds of nodes, this is a minimal list at the time
 | Tapyrus Core  | min 1, unbounded  | Tapyrus full node implementation |
 | Tapyrus Seeder | min 1, unbounded  | DNS seeder for configuring Tapyrus network |
 
-
 ### Tapyrus network parameters
 Following parameters are unique to each tapyrus network
```
Comment (azuchi) [inline doc/tapyrus/getting_started.md]: Please add description of the startup method and environment specifying the networkid.

File: src/chainparams.h [hunk 3]
```diff
@@ -66,60 +61,51 @@ class CChainParams
...
-     * Parse aggPubkey in block header.
-     */
-    CPubKey ReadAggregatePubkey(const std::vector<unsigned char>& pubkey);
-    const CPubKey& GetAggregatePubkey() const{ return aggregatePubkey; }
-    const CBlock& GenesisBlock() const { return genesis; }
+
     /** Default value for -checkmempool and -checkblockindex argument */
     bool DefaultConsistencyChecks() const { return fDefaultConsistencyChecks; }
     /** Policy: Filter transactions that do not match well-defined patterns */
     bool RequireStandard() const { return fRequireStandard; }
```
[the remark this answers is not in the document]
Comment (Naviabheeman) [reply src/chainparams.h]: ok, I'll remove the flag and always accept only standard transactions.

File: src/test/util_tests.cpp (hunk 1 above)
Comment (Naviabheeman) [reply src/test/util_tests.cpp]: I want to verify that the parameter "-testnet" does not have any effect on GetChainMode(). This test is trying to check that parsing no argument and parsing -testnet gives the same chain mode, MAIN.

Comment (azuchi) [reply src/test/util_tests.cpp]: I see. it's ok.

**Comment by Naviabheeman** [comment]
> The following code for testnet remains in `tapyrus-cli.cpp`.
> 
> ```c++
> result.pushKV ("testnet", UniValue (batch [ID_BLOCKCHAININFO] ["result"] ["chain"]. get_str () == "test"));
> ```

I missed this usage completely. I'll check  and correct it.

**Comment by azuchi** [comment]
> > The following code for testnet remains in `tapyrus-cli.cpp`.
> > ```c++
> > result.pushKV ("testnet", UniValue (batch [ID_BLOCKCHAININFO] ["result"] ["chain"]. get_str () == "test"));
> > ```
> 
> I missed this usage completely. I'll check and correct it.

Instead of remove, how about show MODE?

**Comment by Naviabheeman** [comment]
This PR now has the Standard transactions changes and updated getting started document.

**Pull request merged**